pennylane/devices/qutrit_mixed/measure.py

```diff
@@ -0,0 +1,301 @@
+
+
+def calculate_probability(
+    measurementprocess: StateMeasurement, state: TensorLike, is_state_batched: bool = False
+) -> TensorLike:
+    """Find the probability of measuring states.
+
+    Args:
+        measurementprocess (StateMeasurement): measurement to apply to the state.
+        state (TensorLike): state to apply the measurement to.
+        is_state_batched (bool): whether the state is batched or not.
+
+    Returns:
+        TensorLike: the result of the measurement.
+    """
+    for op in measurementprocess.diagonalizing_gates():
+        state = apply_operation(op, state, is_state_batched=is_state_batched)
+
+    num_state_wires = get_num_wires(state, is_state_batched)
+
+    # probs are diagonal elements
+    # stacking list since diagonal function axis selection parameter names
+    # are not consistent across interfaces
+    reshaped_state = reshape_state_as_matrix(state, num_state_wires)
+    if is_state_batched:
+        probs = math.real(math.stack([math.diagonal(dm) for dm in reshaped_state]))
+    else:
+        probs = math.real(math.diagonal(reshaped_state))
+
+    # if a probability is very small it may round to negative, undesirable.
+    # math.clip with None bounds breaks with tensorflow, using this instead:
+    probs = math.where(probs < 0, -probs, probs)
+
+    if mp_wires := measurementprocess.wires:
+        expanded_shape = [QUDIT_DIM] * num_state_wires
+        new_shape = [QUDIT_DIM ** len(mp_wires)]
+        if is_state_batched:
+            batch_size = probs.shape[0]
+            expanded_shape.insert(0, batch_size)
+            new_shape.insert(0, batch_size)
+        wires_to_trace = tuple(
+            x + is_state_batched for x in range(num_state_wires) if x not in mp_wires
+        )
+
+        expanded_probs = math.reshape(probs, expanded_shape)
+        summed_probs = math.sum(expanded_probs, axis=wires_to_trace)
+        return math.reshape(summed_probs, new_shape)
+
+    return probs
+
+
+def calculate_variance(
+    measurementprocess: StateMeasurement, state: TensorLike, is_state_batched: bool = False
+) -> TensorLike:
+    """Find variance of observable.
+
+    Args:
+        measurementprocess (StateMeasurement): measurement to apply to the state.
+        state (TensorLike): state to apply the measurement to.
+        is_state_batched (bool): whether the state is batched or not.
+
+    Returns:
+        TensorLike: the result of the measurement.
+    """
+    probs = calculate_probability(measurementprocess, state, is_state_batched)
+    eigvals = math.asarray(measurementprocess.eigvals(), dtype="float64")
+    # In case of broadcasting, `probs` has two axes and these are a matrix-vector products
+    return math.dot(probs, (eigvals**2)) - math.dot(probs, eigvals) ** 2
+
+
+# pylint: disable=too-many-return-statements
+def get_measurement_function(
+    measurementprocess: MeasurementProcess,
+    state: TensorLike,  # pylint: disable=unused-argument
```

Comment:
```suggestion
```

Is this just an artifact from copying over from `default.qubit`?

Reply:
Yes I think I left a comment about it, I was thinking I should remove it but kept it to check.